xskillscore/core/comparative.py

```diff
@@ -0,0 +1,151 @@
+import warnings
+
+import scipy.stats as st
+import xarray as xr
+
+
+def sign_test(
+    observation,
+    forecast1,
+    forecast2,
+    dim=None,
+    alpha=0.05,
+    metric=None,
+    orientation='negative',
+):
+    """
+        Returns the Delsole and Tippett sign test over the given time dimension.
+
+        Parameters
+        ----------
+        observation : xarray.Dataset or xarray.DataArray or None
+            observation to be compared to both forecasts.
+            If ``None``, then assume that forecast1 and forecast2 have already been
+            compared to observation. If metric is None, this assumes forecasts to be
+            already compared to observation before and ignores observation. Please
+            adjust ``orientation`` accordingly.
+        forecast1 : xarray.Dataset or xarray.DataArray
+            forecast1 to be compared to observation
+        forecast2 : xarray.Dataset or xarray.DataArray
+            forecast2 to be compared to observation
+        dim : str
+            time dimension of dimension over which to compute the random walk.
+            This dimension is not reduced, unlike in other xskillscore functions.
+        alpha : float
+            significance level for random walk.
+        metric : callable, optional
+            metric to compare forecast# with observation if metric is not None. If
+            metric is None, assume that forecast# have been compared observation before
+            using ``sign_test`` and adjust ``orientation``.
```

Comment:
The language is a bit confusing here with "and adjust ``orientation``." Maybe put a period after ``sign_test`` and say "Make sure to adjust ``orientation`` if ``metric`` is ``None``."